public/samples/DataStreams/StreamsUpkeep.sol

```diff
@@ -110,65 +98,48 @@ contract StreamsUpkeep is ILogAutomation, StreamsLookupCompatibleInterface {
         return (true, abi.encode(values, extraData));
     }
 
-    function performUpkeep(bytes calldata performData) external override {
+    // function will be performed on-chain
+    function performUpkeep(bytes calldata performData) external {
+        // Decode incoming performData
         (bytes[] memory signedReports, bytes memory extraData) = abi.decode(
             performData,
             (bytes[], bytes)
         );
 
         bytes memory report = signedReports[0];
 
-        bytes memory bundledReport = bundleReport(report);
+        (, bytes memory reportData) = abi.decode(report, (bytes32[3], bytes));
 
-        BasicReport memory unverifiedReport = _getReportData(report);
+        // Billing
```

Comment:
Instead of billing whichis how we look at it, perhaps phrase from the user POV -- "Report fees"

Reply:
Done